src/main/java/io/confluent/idesidecar/restapi/models/graph/RealDirectFetcher.java

```diff
@@ -106,64 +106,72 @@ public List<DirectConnection> getConnections() {
+            .map(clusterId ->
+              new DirectKafkaCluster(
+                  clusterId,
+                  null,
+                  kafkaConfig.bootstrapServers(),
+                  connectionId
+              )
+            ).map(cluster -> {
+              // Emit an event that this cluster was loaded
+              onLoad(connectionId, cluster);
+              // And return the cluster
+              return cluster;
+            });
+      }).orElseGet(() -> Uni.createFrom().nullItem());
     }
+    // Unexpectedly not a direct connection
+    Log.errorf("Connection with ID=%s is not a direct connection.", connectionId);
     return Uni.createFrom().nullItem();
   }
 
   public Uni<DirectSchemaRegistry> getSchemaRegistry(String connectionId) {
-    var spec = connections.getConnectionSpec(connectionId);
-    DirectSchemaRegistry cluster = null;
-    if (spec != null && DIRECT.equals(spec.type()) && spec.schemaRegistryConfig() != null) {
-      var srConfig = spec.schemaRegistryConfig();
-      var ccloudEndpoint = srConfig.asCCloudEndpoint();
-      if (ccloudEndpoint.isPresent()) {
-        var endpoint = ccloudEndpoint.get();
-        cluster = new DirectSchemaRegistry(
-            endpoint.clusterId().toString(),
-            endpoint.getUri().toString(),
-            connectionId
-        );
-      } else {
-        cluster = new DirectSchemaRegistry(
-            orDefault(srConfig.id(), () -> connectionId + "-schema-registry"),
-            srConfig.uri(),
-            connectionId
-        );
-      }
-    }
-    if (cluster != null) {
-      onLoad(connectionId, cluster);
-      return Uni.createFrom().item(cluster);
+    var state = connections.getConnectionState(connectionId);
+    if (state instanceof DirectConnectionState directState) {
+      var spec = state.getSpec();
+      // Use the admin API to obtain the cluster ID
+      return directState.withSchemaRegistryClient(srClient -> {
+        Uni<DirectSchemaRegistry> result;
+        try {
+          var mode = srClient.getMode(); // unused, but we know we can connect
+          var srConfig = spec.schemaRegistryConfig();
+          var ccloudEndpoint = srConfig.asCCloudEndpoint();
+          DirectSchemaRegistry cluster;
+          if (ccloudEndpoint.isPresent()) {
+            var endpoint = ccloudEndpoint.get();
+            cluster = new DirectSchemaRegistry(
+                endpoint.clusterId().toString(),
+                endpoint.getUri().toString(),
+                connectionId
+            );
+          } else {
+            cluster = new DirectSchemaRegistry(
+                orDefault(srConfig.id(), () -> connectionId + "-schema-registry"),
+                srConfig.uri(),
+                connectionId
+            );
+          }
+          result = Uni.createFrom().item(cluster);
+        } catch (Throwable t) {
+          Log.debugf("Failed to create schema registry client for connection ID=%s", connectionId, t);
```

Comment:
Shouldn't this say something like "failed to connect"? Technically at this point we have an SR client instance that was used to make the global `/mode` API call.

```suggestion
          Log.debugf("Failed to connect to schema registry for connection ID=%s", connectionId, t);
```

Reply:
Improved the error messages.